src/title_screen.c

```diff
@@ -0,0 +1,1056 @@
+{
+    u8 spriteId;
+    if (templateId)
+    {
+        spriteId = CreateSprite(&sSpriteTemplate_FlameOrLeaf_State1, x, y, 0);
+    }
+    else
+    {
+        spriteId = CreateSprite(&sSpriteTemplate_FlameOrLeaf_State0, x, y, 0);
+    }
+    if (spriteId != MAX_SPRITES)
+    {
+        gSprites[spriteId].data[0] = x << 4;
+        gSprites[spriteId].data[1] = xspeed;
+        gSprites[spriteId].data[2] = y << 4;
+        gSprites[spriteId].data[3] = yspeed;
+        gSprites[spriteId].data[4] = 0;
+        gSprites[spriteId].data[5] = (xspeed * yspeed) % 16;
+        gSprites[spriteId].data[6] = templateId;
+        gSprites[spriteId].callback = SpriteCallback_TitleScreenFlameOrLeaf;
+        return TRUE;
+    }
+    return FALSE;
+}
+
+static void Task_FlameOrLeafSpawner(u8 taskId)
+{
+    s16 * data = gTasks[taskId].data;
+    s32 x, y, xspeed, yspeed, templateId;
+    s32 i;
+
+    switch (data[0])
+    {
+    case 0:
+        TitleScreen_srand(taskId, 3, 30840);
+        data[0]++;
+        break;
+    case 1:
+        data[1]++;
+        if (data[1] >= data[2])
+        {
+            data[1] = 0;
+            TitleScreen_rand(taskId, 3);
+            data[2] = 18;
+            xspeed = (TitleScreen_rand(taskId, 3) % 4) - 2;
+            yspeed = (TitleScreen_rand(taskId, 3) % 8) - 16;
+            y = (TitleScreen_rand(taskId, 3) % 3) + 0x74;
+            x = TitleScreen_rand(taskId, 3) % 240;
+            CreateFlameOrLeafSprite(
+                x,
+                y,
+                xspeed,
+                yspeed,
+                (TitleScreen_rand(taskId, 3) % 16) < 8 ? 0 : 1
+            );
+            for (i = 0; i < 15; i++)
+            {
+                CreateFlameOrLeafSprite(
+                    data[5] + gUnknown_83BFBD4[i],
+                    y,
+                    xspeed,
+                    yspeed,
+                    1
+                );
+                xspeed = (TitleScreen_rand(taskId, 3) % 4) - 2;
+                yspeed = (TitleScreen_rand(taskId, 3) % 8) - 16;
+            }
+            data[5]++;
+            if (data[5] > 3)
+                data[5] = 0;
+        }
+    }
+}
+
+static void TitleScreen_srand(u8 taskId, u8 field, u16 seed)
+{
+    SetWordTaskArg(taskId, field, seed);
+}
+
+static u16 TitleScreen_rand(u8 taskId, u8 field)
+{
+    u32 rngval;
+
+    rngval = GetWordTaskArg(taskId, field);
+    rngval = rngval * 1103515245 + 24691;
+    SetWordTaskArg(taskId, field, rngval);
+    return rngval >> 16;
+}
+
+static u32 CreateBlankSprite(void)
+{
+    CreateSprite(&sUnknownEmptySprite, 0x18, 0x90, 0);
+    return IndexOfSpritePaletteTag(2);
+}
+
+static void SetPalOnOrCreateBlankSprite(bool32 mode)
+{
+    u32 palIdx;
+
+    if (mode)
+    {
+        palIdx = IndexOfSpritePaletteTag(2);
+        LoadPalette(gGraphics_TitleScreen_FirePals, palIdx * 16 + 0x100, 0x20);
+    }
+    else
+        CreateBlankSprite();
```

Comment:
A chain of if-else statements in which any block is more than one line of code should use braces.  - from style guide

Reply:
Resolving without change since the guide is just that -- a guide, rather than hard-and-set rules. There is no evidence that GameFreak followed that style.